src/test/components/elements/panels/ManageExistingBookings.test.tsx

```diff
@@ -0,0 +1,183 @@
+      firstUser.gender || "",
+      FRIENDLY_DATE_AND_TIME.format(new Date(mockEventBookingsForMockEvent[0].bookingDate!)),
+      FRIENDLY_DATE_AND_TIME.format(new Date(mockEventBookingsForMockEvent[0].updated!)),
+      Array.from(new Set(firstUser.registeredContexts?.map((rc) => stageLabelMap[rc.stage!]))).join(", "),
+      Array.from(new Set(firstUser.registeredContexts?.map((rc) => examBoardLabelMap[rc.examBoard!]))).join(", "),
+      mockEventBookingsForMockEvent[0].reservedById ? mockEventBookingsForMockEvent[0].reservedById.toString() : "-",
+      mockEventBookingsForMockEvent[0].additionalInformation!.accessibilityRequirements || "",
+      mockEventBookingsForMockEvent[0].additionalInformation!.medicalRequirements || "",
+      mockEventBookingsForMockEvent[0].additionalInformation!.emergencyName || "",
+      mockEventBookingsForMockEvent[0].additionalInformation!.emergencyNumber || "",
+    ];
+    expectRowData(firstBookingRow, expectedFirstBookingData);
+  });
+
+  it("displays a message if no bookings are found for the event being checked", async () => {
+    setupTest([]);
+    expect(screen.getByText("Manage current bookings")).toBeInTheDocument();
+    const noEventsMessage = await screen.findByText(/There aren't currently any bookings for this event/i);
+    expect(noEventsMessage).toBeInTheDocument();
+  });
+
+  it("displays action buttons for each booking", async () => {
+    setupTest(mockEventBookingsForMockEvent);
+    const { firstBookingRow } = await getTableAndFirstRow();
+    const firstBookingCells = within(firstBookingRow).getAllByRole("cell");
+    const firstBookingActionsCell = firstBookingCells[0];
+    const actionButtons = within(firstBookingActionsCell).getAllByRole("button");
+    expect(actionButtons).not.toHaveLength(0);
+  });
+
+  const deleteButtonTestCases = [
+    { role: "ADMIN" as UserRole, description: "offers", expected: true },
+    { role: "EVENT_LEADER" as UserRole, description: "does not offer", expected: false },
+    { role: "EVENT_MANAGER" as UserRole, description: "does not offer", expected: false },
+  ];
+
+  it.each(deleteButtonTestCases)("$description delete button for $role", async ({ role, expected }) => {
+    setupTest(mockEventBookingsForMockEvent, role);
+    const { firstBookingRow } = await getTableAndFirstRow();
+    const deleteButton = await getActionButton(firstBookingRow, "delete");
+    if (expected) {
+      expect(deleteButton).toBeInTheDocument();
+    } else {
+      expect(deleteButton).toBeNull();
+    }
+  });
+
+  const buttonTestCases = [
+    { buttonName: "promote", status: ["WAITING_LIST", "CANCELLED"] as BookingStatus[] },
+    { buttonName: "cancel", status: ["WAITING_LIST", "CONFIRMED"] as BookingStatus[] },
+    {
+      buttonName: "resend email",
+      status: ["WAITING_LIST", "CONFIRMED", "CANCELLED", "ATTENDED", "ABSENT", "RESERVED"] as BookingStatus[],
+    },
+  ];
+
+  buttonTestCases.forEach(({ buttonName, status }) => {
```

Comment:
You're not supposed to do a `foreEach` here. Instead, you're supposed to pass the array to `it.each`.

Reply:
I was trying to achieve two loops here, so it checked each button, but then within that it checked each booking status specified.  I've just refactored this to simplify it and just give a longer list of test cases and a single `it.each` instead.